nixos/modules/services/security/endlessh-go.nix

```diff
@@ -0,0 +1,136 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.endlessh-go;
+in
+{
+  options.services.endlessh-go = {
+    enable = mkEnableOption "endlessh-go service";
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      example = "[::]";
```

Comment:
Why do we by default not listen on ipv6?

Reply:
I kept the upstream [default](https://github.com/shizunge/endlessh-go#usage):
```
  -host string
        SSH listening address (default "0.0.0.0")
```

Reply:
Did you test if ``[::]`` works at all? Maybe 0.0.0.0 listens on both protocols? If not then I would default to include ipv6 nevertheless what the upstream default is.

Reply:
It works, but the TCP version binding is managed by the `-conn_type` which can be either `tcp`, `tcp4` or `tcp6`. I was assuming that `tcp` should bind to both versions but it only builds to tcp4 (I guess this is an upstream bug). In short, if you want to bind to `[::]` you need to also pass `-conn_type=tcp6`.

Since this service is a honeypot, and since we can't allow multiple binds, I think the sensible approach would be to bind to ipv4 by default assuming that tcp4 will get more traffic.